pyanaconda/ui/gui/spokes/root_password.py

```diff
@@ -216,7 +219,14 @@ def apply(self):
 
     @property
     def completed(self):
-        return self._users_module.IsRootPasswordSet
+        """Is the spoke completed?
+
+        For root and user, the mandatory+completed pair is a complicated hack. Having an usable
+        admin user is mandatory, but it is not clear if it should be an unlocked root, or a sudoer.
+        Thus, mandatory on both spokes checks admin user, and complete then checks again: Both the
+        spoke-specific completion condition, as well as existence of an admin.
+        """
+        return self._users_module.IsRootPasswordSet and self._users_module.CheckAdminUserExists()
```

Comment:
I think that the real issue is that the root password is still set to the previous value after the user disables the root account.

Reply:
Yes, that would solve the problem. But that is arguably a feature. I think setting password for locked users is a thing - correct me if I'm wrong.

Reply:
I wouldn't expect that behaviour based on the current UI. Locked root with a password is supported with kickstart, but I thought we don't want to support that in the UI. What do others think? @jkonecny12, @M4rtinK

Reply:
I agree that it's not something we want to support in GUI. In general it could be confusing for people.

Reply:
IIRC the previous GUI "supported" it, so it will be a change. But I very much like the idea, it simplifies things.

Reply:
Agreed with not supporting setting a root password in GUI and then locking it - that really seems pretty esoteric to me & non-issue for advanced users if supported in kickstart.

Reply:
Not to mention an admin level user account or service running with root privileges will be needed to unlock the root password. That thing can also set the password to a value if needed.

Reply:
Fixed at https://github.com/rhinstaller/anaconda/pull/3775.